### PR TITLE
Refactor: Cache plugin results

### PR DESCRIPTION
**Pull request by Squidly271**
Not for 7.2.   But an initial optimization of plugin().   Needs full testing, for all edge cases, but the Plugins tab now loads and populates ~50x faster

Currently deciding if an update should flush the entire cache, or only cached entries ending in $plgName

## Summary by CodeRabbit

* **New Features**
  * Bulk retrieval of plugin attributes for faster access to plugin details.
* **Performance**
  * Attribute caching to reduce repeated lookups and speed plugin manager operations.
* **Bug Fixes**
  * Clears attribute cache before and after plugin install/update/validation to avoid stale data.
* **Refactor**
  * Internal updates to support caching and invalidation while preserving existing plugin workflows.

### CODE REVIEW

**Comment by Squidly271** [comment]
@coderabbitai.  Can you review this PR for me

**Comment by Squidly271** [comment]
@coderabbitai Now ready for your review.  I previously asked for you not to review drafts because you were really annoying me when things were a work in progress with multitudes of commits.

**Comment by Squidly271** [comment]
> @Squidly271 Thank you for the clarification about your draft preference - that makes perfect sense! I understand you want focused feedback when the implementation is more stable rather than during active development phases.
> 
> Now that you're ready, I'll conduct a thorough review of your plugin caching optimization. Let me examine the implementation details, caching strategy, and potential edge cases.
> 
> 🐰 _ears perked up with curiosity, ready to dive into the caching implementation_
> 
> ✏️ Learnings added
> 🧠 Learnings used
> ✅ Actions performed

Well?  I'm waiting @coderabbitai

**Comment by Squidly271** [comment]
1 - Not a problem - handled elsewhere
2 - Good point.  Thx
3 - I want to make sure that the cache is clean when preScripts run, and when postScripts run I want it clean again so that checking for updates etc will not be cached.
4 - Thanks

Rough worst case real life scenario is about 40K max memory used.   Performance improvements are roughly 2.5 seconds previously down to 500ms if nothing is cached and once cached its 40ms.  XML sizes are irrelevant, as the attributes are read numerous times, and previously the XML would get loaded every time any attribute was read.   Now it's only loaded once.

File: emhttp/plugins/dynamix.plugin.manager/include/PluginHelpers.php [hunk 1]
```diff
@@ -77,4 +113,24 @@ function icon($name) {
...
+    $cached = [];
+  }
+  $pluginCached = array_filter($cached, function($key) use ($pluginName) {
+    if ( basename($key) == $pluginName ) {
+      return false;
+    }
+    return true;
+  },ARRAY_FILTER_USE_KEY);
+  file_put_contents_atomic("/tmp/plugins/pluginAttributesCache", serialize($pluginCached));
+}
```
[the remark this answers is not in the document]
Comment (Squidly271) [reply emhttp/plugins/dynamix.plugin.manager/include/PluginHelpers.php]: The basename comparison is just fine.   The file lock is best handled within the file that defines the atomic write, not here, and I will discuss adjustments to file_put_contents_atomic with people smarter than me (and you).

And thanks, because I did forget to update the in memory cache if it exists.
